mlir/include/mlir/Dialect/Tosa/IR/TosaProfileCompliance.h

```diff
@@ -0,0 +1,163 @@
+//===- TosaProfileCompliance.h - Tosa Profile-based Compliance Validation -===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_TOSA_TRANSFORMS_TOSAPROFILECOMPILANCE_H
+#define MLIR_DIALECT_TOSA_TRANSFORMS_TOSAPROFILECOMPILANCE_H
+
+#include "mlir/Dialect/Tosa/IR/TargetEnv.h"
+#include "mlir/Dialect/Tosa/Transforms/Passes.h"
+
+#include "mlir/Support/TypeID.h"
+
+using namespace mlir;
+using namespace mlir::tosa;
+
+//===----------------------------------------------------------------------===//
+// Type Compilance Definition
+//===----------------------------------------------------------------------===//
+
+typedef struct {
+  mlir::TypeID typeID;
+  uint32_t bitWidth;
+} TypeInfo;
+
+enum CheckCondition {
+  // Valid when any of the profile (extension) requirement is meet.
+  anyOf,
+  // Valid when all of the profile (extension) requirement are meet.
+  allOf,
+  invalid
+};
+
+template <typename T>
+struct OpComplianceInfo {
+  // Certain operations require multiple modes enabled.
+  // e.g. cast bf16 to fp8e4m3 requires EXT-BF16 and EXT-FP8E4M3.
+  SmallVector<T> mode;
+  SmallVector<SmallVector<TypeInfo>> operandTypeInfoSet;
+  CheckCondition condition = CheckCondition::anyOf;
+};
+
+using OperationProfileComplianceMap =
+    std::unordered_map<std::string, SmallVector<OpComplianceInfo<Profile>>>;
```

Comment:
[`std::unordered_map`](https://en.cppreference.com/w/cpp/container/unordered_map) requires the `unordered_map` header file